### DISCUSSION TITLE
environment variables for jupyter notebook

### DISCUSSION THREAD

**Discussion by mohit2152sharma** [microsoft/vscode-jupyter#9739] [Questions and Answers] [answered]
In vscode `settings.json` file I can use the following option to define environment variables:
``` json
"terminal.integrated.env.osx" : {
    "MY_ENV": "test"
    "MY_ENVTYPE": "qa"
}
```
Now whenever, I start a new shell in the workspace, the shell loads with the above environment variables, and I can access them typically with `os.environ["MY_ENV"]` is my python scripts.

But with the same `settings.json`, if I try to access the environment variables in a jupyter notebook I get `None`. So my question is, is there a way to define environment variables in vscode's `settings.json` file, so whenever I start a new notebook, the environment variables are loaded by default. 

Currently the workaround I have found is to add the following code snippet in a top code cell.

``` python

import os
os.environ["MY_ENV"] = "test"
os.environ["MY_ENVTYPE"] = "qa"
```
I am hoping there is a better way to do the same.

**Comment by IanMatthewHuff** [comment, marked as answer]
We actually have a setting that can run some arbitrary code at the start up of a any notebook or interactive window. Give this a try, should be what you need. If you just put that code snippet into the setting.
<img width="1254" alt="image" src="https://user-images.githubusercontent.com/812783/164514211-73501554-0505-41f6-9a93-5e7b29bbe1f4.png">
